Src/Particle/AMReX_ParticleContainerI.H

```diff
@@ -60,10 +64,51 @@ ParticleContainer_impl<ParticleType, NArrayReal, NArrayInt, Allocator, CellAssig
         pp.query("do_unlink", doUnlink);
         pp.queryAdd("do_mem_efficient_sort", memEfficientSort);
 
+        // add default names for SoA Real and Int compile-time arguments
+        int first_r_name = 0;
+        // push back x,y,z
+        if constexpr (ParticleType::is_soa_particle) {
+            constexpr int x_in_ascii = 120;
+            for (int i=0; i<AMREX_SPACEDIM; ++i)
+            {
+                std::string const name{char(x_in_ascii+i)};
+                m_soa_rdata_names.push_back(name);
+            }
+            first_r_name = AMREX_SPACEDIM;
+        }
+        for (int i=first_r_name; i<NArrayReal; ++i)
+        {
+            m_soa_rdata_names.push_back("real_comp" + std::to_string(i-first_r_name));
+        }
+        for (int i=0; i<NArrayInt; ++i)
+        {
+            m_soa_idata_names.push_back("int_comp" + std::to_string(i));
+        }
+
```

Comment:
We discussed that we want to make this two functions (one for real, one for int) that take the ParticleType and index to return a default name on the current index.

This way, we can reuse this in the:
- current I/O logic (existing naming scheme we took over here)
- here
- in `AddRealComp`/`AddIntComp` without `name` argument